lib/rspec/buildkite/analytics/session.rb

```diff
@@ -177,11 +172,53 @@ def connect
       wait_for_confirm
 
       @logger.write("connected")
+
+      # As this connect method can be called multiple times in the
+      # reconnection process, kill prev write threads (if any) before
+      # setting up the new one
+      @write_thread&.kill
+
+      @write_thread = Thread.new do
+        @logger.write("hello from write thread")
+        # Pretty sure this while loop is fine cos the call to queue.pop is blocking
+        while
+          data = @send_queue.pop
+          message_type = data["action"]
+
+          if message_type == "end_of_transmission"
+            # Because of the unpredictable sequencing between the test suite finishing
+            # (EOT gets queued) and disconnections happening (retransmit results gets
+            # queued), we don't want to send an EOT before any retransmits are sent.
+            unless @send_queue.empty?
```

Comment:
For this I find `if any?` easier to follow.
```suggestion
            if @send_queue.any?
```

Reply:
Unfortunately this is a Queue and not an Enumerable so this method isn't available! https://ruby-doc.org/core-2.5.0/Queue.html

Reply:
what about?
```suggestion
            if @send_queue.length > 0
```